package.json

```
@@ -1,10 +1,10 @@
 {
   "name": "vanilla-spa-nodejs-single-file",
   "version": "1.0.0",
-  "description": "",
-  "main": "index.js",
+  "description": "vanilla SPA with webpack",
   "scripts": {
-    "start": "node index.js"
+    "build": "webpack",
+    "start": "npm run build && node src/server/index.js"
```

Ini perlu implement dev mode, tapi butuh nambah `nodemon` dan `concurrently`

```suggestion
    "build": "webpack",
    "start": "npm run build && node src/server/index.js",
    "dev": "concurrently \"npm run build -- --watch\"  \"nodemon src/server/index.js\"",
```

---

Kalau mau lebih rapih lagi, bisa pake file config yang berbeda, jadinya kayak gini : 

```
"build:prod": "webpack --config webpack.prod.js",
"build:dev": "webpack --config webpack.dev.js",
"start": "node src/server/index.js",
"dev": "concurrently \"npm run build:dev\"  \"nodemon src/server/index.js\"",
```

---

Cara setting mode di config js webpack : 
https://webpack.js.org/configuration/mode/

Cara setting watch di config js webpack : 
https://webpack.js.org/configuration/watch/

---

Jadi kamu bakalan butuh bikin dua file : `webpack.dev.js` dan `webpack.prod.js`, untuk menghindari duplikasi config, kamu bisa bikin satu base file, misalnya `webpack.base.js`, dan di extend ke config prod dan dev pake : https://webpack.js.org/configuration/extending-configurations/